cmd/metal-api/internal/grpc/server.go

```diff
@@ -0,0 +1,81 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	v1 "github.com/metal-stack/metal-api/pkg/api/v1"
+	"github.com/metal-stack/metal-lib/zapup"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+	"io/ioutil"
+	"net"
+	"time"
+)
+
+func Serve(ws *WaitServer) {
+	logger := zapup.MustRootLogger().Sugar()
+
+	grpcPort := viper.GetInt("grpc-port")
+	addr := fmt.Sprintf(":%d", grpcPort)
+
+	kaep := keepalive.EnforcementPolicy{
+		MinTime:             5 * time.Second, // If a client pings more than once every 5 seconds, terminate the connection
+		PermitWithoutStream: true,            // Allow pings even when there are no active streams
+	}
+	kasp := keepalive.ServerParameters{
+		Time:    5 * time.Second, // Ping the client if it is idle for 5 seconds to ensure the connection is still active
+		Timeout: 1 * time.Second, // Wait 1 second for the ping ack before assuming the connection is dead
+	}
+	grpcServer := grpc.NewServer(
+		grpc.KeepaliveEnforcementPolicy(kaep),
+		grpc.KeepaliveParams(kasp),
+	)
+	v1.RegisterWaitServer(grpcServer, ws)
+
+	tlsConfig := &tls.Config{
+		NextProtos: []string{"h2"},
+	}
+	if viper.GetBool("grpc-tls-enabled") {
+		cert, err := ioutil.ReadFile(viper.GetString("grpc-server-cert-file"))
+		if err != nil {
+			logger.Fatalw("failed to serve gRPC", "error", err)
```

Comment:
I think, returning an error would be a little more elegant. We should try to avoid `Fatal` and `panic` in the code as much as possible.